tests/test_magic.py

```diff
@@ -914,6 +914,60 @@ def test_notebook_export_json():
         _ip.run_line_magic("notebook", "%s" % outfile)
 
 
+def test_notebook_export_json_with_output():
+    """Tests if notebook export correctly captures outputs, errors, display outputs, and stream outputs."""
+    pytest.importorskip("nbformat")
+    import nbformat
+
+    _ip = get_ipython()
+    _ip.history_manager.reset()
+
+    cmds = [
+        "display('test')",
+        "1+1",
+        "1/0",
+        "print('test')",
+        "1",
```

Comment:
It looks current implementation does not cover the stderr. Let's add it to the test. Also, 1 and 1+1 test the same thing right?

```suggestion
        "import sys",
        "print('test stderr', file=sys.stderr)"
```